src/dowel/csv_output.py

```diff
@@ -30,26 +32,41 @@ def record(self, data, prefix=''):
         if isinstance(data, TabularInput):
             to_csv = data.as_primitive_dict
 
-            if not to_csv.keys() and not self._writer:
+            # if not to_csv.keys() and not self._writer:
+            #     return
+
+            # if not self._writer:
+            #     self._fieldnames = set(to_csv.keys())
+            #     self._writer = csv.DictWriter(
+            #         self._log_file,
+            #         fieldnames=self._fieldnames,
+            #         extrasaction='ignore')
+            #     self._writer.writeheader()
+
+            # if to_csv.keys() != self._fieldnames:
+            #     self._warn('Inconsistent TabularInput keys detected. '
+            #                'CsvOutput keys: {}. '
+            #                'TabularInput keys: {}. '
+            #                'Did you change key sets after your first '
+            #                'logger.log(TabularInput)?'.format(
+            #                    set(self._fieldnames), set(to_csv.keys())))
+
+            # self._writer.writerow(to_csv)
+
+            if not to_csv.keys():
                 return
 
-            if not self._writer:
-                self._fieldnames = set(to_csv.keys())
-                self._writer = csv.DictWriter(
-                    self._log_file,
-                    fieldnames=self._fieldnames,
-                    extrasaction='ignore')
-                self._writer.writeheader()
-
-            if to_csv.keys() != self._fieldnames:
-                self._warn('Inconsistent TabularInput keys detected. '
-                           'CsvOutput keys: {}. '
-                           'TabularInput keys: {}. '
-                           'Did you change key sets after your first '
-                           'logger.log(TabularInput)?'.format(
-                               set(self._fieldnames), set(to_csv.keys())))
-
-            self._writer.writerow(to_csv)
+            # Add new data, Pandas will create new columns automatically
+            self._df = self._df.append(to_csv, ignore_index=True)
```

Comment:
What does the `ignore_index` parameter do? Why is it needed here?

Reply:
Pandas DataFrame will add the index like 0-n for each row in CSV, setting ignore_index=True will prevent creating the extra column of each row.